phoenix-core-client/src/main/java/org/apache/phoenix/util/IndexUtil.java

```diff
@@ -856,6 +856,40 @@ public static void removeEmptyColumn(Mutation m, byte[] emptyCF, byte[] emptyCQ)
         }
     }
 
+    /**
+     * Creates a true deep copy of the Put Mutation, including deep copies of all cells if the
+     * cells are backed by off-heap. The Mutation(Mutation source) constructor always does a
+     * shallow copy of the cells.
+     * @param original The original Put Mutation to copy
+     * @return A new Put Mutation with deep copies of all fields and Cells
+     */
+    public static Put copyPut(Put original) throws IOException {
```

Comment:
While this might only be used by IndexRegionObserver for now, this can be used by others also in the future. How about keeping it as separate utility class e.g. `CellUtil`?